### PR TITLE
Raise ValueError in SIFT when sigma_in >= sigma_min / upsampling

### LINKED ISSUE #8303: SIFT crashes with ValueError when sigma_in exceeds sigma_min
### What's happening

`SIFT.detect_and_extract` crashes with a `ValueError` when `sigma_in`
(the assumed blur of the input image) is set higher than `sigma_min`
(the seed blur level, default `1.6`). The docstring doesn't mention
any relationship between these two parameters, so nothing warns that
`sigma_in > sigma_min` is invalid.

### How to reproduce it

```python
import numpy as np
from skimage.feature import SIFT

img = np.zeros((100, 100))
extractor = SIFT(sigma_in=2.0)
extractor.detect_and_extract(img)
```

### What actually happens

On the officially released package (via `pip install scikit-image`):

On a development build, the same crash surfaces with a more
descriptive message:

Both come from the same line — `math.sqrt(self.sigma_min**2 -
self.sigma_in**2)` going negative.

### Why it fails

- `sigma_min` defaults to `1.6`, `sigma_in` defaults to `0.5`
- In `_create_scalespace`, the amount of additional blur to apply is
  computed as:
  `math.sqrt(self.sigma_min**2 - self.sigma_in**2)`
- This is the classic "combine two Gaussian blurs" formula: if the
  input is already blurred by `sigma_in`, only `sqrt(sigma_min² -
  sigma_in²)` more blur is needed to reach `sigma_min`
- That only makes sense when `sigma_in <= sigma_min`. When `sigma_in`
  exceeds `sigma_min` (e.g. `sigma_in=2.0` with the default
  `sigma_min=1.6`), the value under the square root goes negative,
  and the crash follows
- Nothing in `SIFT.__init__` or `_create_scalespace` validates this
  relationship

### Evidence this is a real, reproducible bug

- Reproduced in two separate environments: a local development build
  and a fresh Google Colab session using the officially released
  `pip install scikit-image` package — same root cause confirmed in
  both, with slightly different `ValueError` messages depending on
  the `sqrt` implementation used
- No existing test in the SIFT test suite touches `sigma_in` at all
- No existing doctest sets `sigma_in` to anything other than the
  default
- The docstring documents `sigma_in` and `sigma_min` independently,
  with no mention that one must not exceed the other

### Suggested fix

Add a validation check in `SIFT.__init__` (or at the point
`sigma_min`/`sigma_in` are used), e.g.:

```python
if sigma_in >= sigma_min:
    raise ValueError('sigma_in must be less than sigma_min')
```

Alternatively, clamp the computed variance difference at zero (similar
to the `max(0, ...)` pattern used elsewhere in this codebase for
inertia-tensor-eigenvalue-derived quantities), if silently skipping
the extra blur step is acceptable behavior instead of raising.

### Version info

scikit-image version: 0.25.2
numpy version: 2.1.3
Platform: Linux-6.6.122+-x86_64-with-glibc2.35

### PR DESCRIPTION
**Pull request by ravikumar266**
Fixes #8303 — SIFT raised an unhandled ValueError (or "math domain
error" on the released package) instead of a clear validation error
when sigma_in was set greater than sigma_min / upsampling.

Root cause
_create_scalespace computes the additional blur needed to reach the
seed level as math.sqrt(self.sigma_min**2 - self.sigma_in**2). This
combine-two-Gaussian-blurs formula only makes sense when
sigma_in <= sigma_min (after upsampling scaling). When sigma_in
exceeds it (e.g. sigma_in=2.0 with the default sigma_min=1.6,
upsampling=2 giving an effective sigma_min of 0.8), the value under
the square root goes negative.

Fix
Added an explicit validation check in SIFT.__init__, right after
self.sigma_min is computed:

if sigma_in > self.sigma_min:
    raise ValueError("sigma_in must not exceed sigma_min / upsampling")

Note the boundary is inclusive — sigma_in == sigma_min / upsampling
is valid, since it just means zero additional blur is needed
(sqrt(0) == 0). This was caught in review (thanks Copilot) and fixed
in a follow-up commit; the original version incorrectly rejected the
boundary case.

Also updated the sigma_in docstring entry to document the constraint.

Testing
Added two regression tests:
- test_sift_sigma_in_exceeds_sigma_min_raises — confirms a clean
  ValueError when sigma_in exceeds the limit
- test_sift_sigma_in_equal_to_sigma_min_is_valid — locks in that the
  boundary itself (sigma_in == sigma_min / upsampling) is accepted,
  not rejected
Confirmed default SIFT() still runs correctly on a real image
(data.camera()), finding keypoints as expected
Ran the full SIFT test suite locally — all tests pass
Ran doctests for the module — pass
Ran pre-commit run --all-files — all checks pass

Note
Used an LLM for research assistance during investigation. All
findings were independently verified locally before being acted on.

### CODE REVIEW

**Comment by ravikumar266** [comment]
Hi @scikit-image/core-devs — I've updated and reopened this PR with a clean single commit (`73aeca5`) addressing the earlier review feedback. All CI checks and pre-commit hooks are green. 

Ready for maintainer review whenever someone has a moment. Thanks!

File: tests/skimage2/feature/test_sift.py [hunk 1]
```diff
@@ -177,3 +177,22 @@ def test_no_descriptors_extracted_sift():
     detector_extractor = SIFT()
     with pytest.raises(RuntimeError):
         detector_extractor.detect_and_extract(img)
+
+
+def test_sift_sigma_in_exceeds_sigma_min_raises():
```
Comment (stefanv) [inline tests/skimage2/feature/test_sift.py:182]: Thanks, @ravikumar266; same as last time, let's not add a test for this guard. The guard itself looks fine.

Comment (ravikumar266) [reply tests/skimage2/feature/test_sift.py]: Done, removed the test. The guard and docstring are unchanged.

**Comment by ravikumar266** [comment]
Updated per @stefanv's review: the regression tests are removed, and the PR now contains only the guard and its docstring change. I've synced with main, and the checks are green apart from the workflows that need maintainer approval. Ready for another look whenever someone has time. Thanks!

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
Thanks!
